packages/SwingSet/test/vat-admin/terminate/test-terminate.js

```diff
@@ -16,83 +16,276 @@ test.before(async t => {
+    false,
+    'done result Error: exceptionallyHappy (Error=true)',
+  );
+});
+
+test('exit happy path complex result (dynamic, critical)', async t => {
+  await doTerminateCritical(t, 'v8', 'exceptionallyHappy', true);
+});
+
+test('exit happy path complex result (static, critical)', async t => {
+  await doTerminateCritical(t, 'v3', 'exceptionallyHappy', false);
+});
+
+test('exit sad path simple result (dynamic, non-critical)', async t => {
+  await doTerminateNonCritical(
+    t,
+    'v8',
+    'sad',
+    true,
+    'done exception sad (Error=false)',
+  );
+});
+
+test('exit sad path simple result (static, non-critical)', async t => {
+  await doTerminateNonCritical(
+    t,
+    'v2',
+    'sad',
+    false,
+    'done exception sad (Error=false)',
+  );
+});
+
+test('exit sad path simple result (dynamic, critical)', async t => {
+  await doTerminateCritical(t, 'v8', 'sad', true);
+});
+
+test('exit sad path simple result (static, critical)', async t => {
+  await doTerminateCritical(t, 'v3', 'sad', false);
 });
 
-test('exit sad path complex result', async t => {
-  await doTerminate(
+test('exit sad path complex result (dynamic, non-critical)', async t => {
+  await doTerminateNonCritical(
     t,
+    'v8',
     'exceptionallySad',
+    true,
     'done exception Error: exceptionallySad (Error=true)',
   );
 });
 
-test('exit happy path with ante-mortem message', async t => {
-  await doTerminate(
+test('exit sad path complex result (static, non-critical)', async t => {
+  await doTerminateNonCritical(
+    t,
+    'v2',
+    'exceptionallySad',
+    false,
+    'done exception Error: exceptionallySad (Error=true)',
+  );
+});
+
+test('exit sad path complex result (dynamic, critical)', async t => {
+  await doTerminateCritical(t, 'v8', 'exceptionallySad', true);
+});
+
+test('exit sad path complex result (static, critical)', async t => {
+  await doTerminateCritical(t, 'v3', 'exceptionallySad', false);
+});
+
+test('exit happy path with ante-mortem message (dynamic, non-critical)', async t => {
+  await doTerminateNonCritical(
+    t,
+    'v8',
+    'happyTalkFirst',
+    true,
+    'done result happyTalkFirst (Error=false)',
+    ['GOT QUERY not dead quite yet'],
+  );
+});
+
+test('exit happy path with ante-mortem message (static, non-critical)', async t => {
+  await doTerminateNonCritical(
     t,
+    'v2',
     'happyTalkFirst',
+    false,
     'done result happyTalkFirst (Error=false)',
     ['GOT QUERY not dead quite yet'],
   );
 });
 
-test('exit sad path with ante-mortem message', async t => {
-  await doTerminate(
+test('exit happy path with ante-mortem message (dynamic, critical)', async t => {
+  await doTerminateCritical(t, 'v8', 'happyTalkFirst', true);
+});
+
+test('exit happy path with ante-mortem message (static, critical)', async t => {
+  await doTerminateCritical(t, 'v3', 'happyTalkFirst', false);
+});
+
+test('exit sad path with ante-mortem message (dynamic, non-critical)', async t => {
```

Comment:
Let's add a comment (on the last argument line of the `doTerminateNonCritical` call) to remind ourselves that `GOT QUERY not dead quite yet (but soon)` should *not* be observed, because `vatPowers.exitVatWithFailure` is explicitly defined to unwind any message sent from the final crank. The test does exactly the right thing (thank you!) but because it's exercising an omission, it's not obvious what property is being checked (and therefore what we must make sure we retain in the future).